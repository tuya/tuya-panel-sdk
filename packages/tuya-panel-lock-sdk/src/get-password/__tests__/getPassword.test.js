import React from 'react';
import { shallow } from 'enzyme';
import GetPassword from '../index';

describe('getPassword components', () => {
  it('basic render', () => {
    const wrapper = shallow(<GetPassword onValueChange={jest.fn()}></GetPassword>);
    expect(wrapper).toMatchSnapshot();
  });
  it('passwordColor render', () => {
    const wrapper = shallow(
      <GetPassword passwordColor={'pink'} onValueChange={jest.fn()}></GetPassword>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('passwordNumber render', () => {
    const wrapper = shallow(
      <GetPassword
        randomText="随机生成"
        passwordNumber={6}
        inputItemText="请输入"
        randomTextColor="#0076FF"
        passwordColor="#0076FF"
        passwordColor={'pink'}
        onValueChange={jest.fn()}
      ></GetPassword>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should TouchableOpacity event', () => {
    const setupThree = (props = {}) => {
      const wrapper = shallow(
        <GetPassword
          randomText="随机生成"
          passwordNumber={6}
          inputItemText="请输入"
          randomTextColor="#0076FF"
          passwordColor="#0076FF"
          passwordColor={'pink'}
          onValueChange={jest.fn()}
        ></GetPassword>
      );
      const instance = wrapper.instance();
      return { wrapper, instance };
    };
    const { wrapper } = setupThree();
    const targetNode = wrapper.findWhere(
      node => node.name() === 'TouchableOpacity' && !!node.prop('onPress') === true
    );
    targetNode.at(0).simulate('press');
    targetNode.at(1).simulate('press');
    targetNode.at(2).simulate('press');
    expect(wrapper).toMatchSnapshot();
  });
});
