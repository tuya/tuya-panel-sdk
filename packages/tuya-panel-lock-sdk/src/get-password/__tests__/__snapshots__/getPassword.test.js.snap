// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPassword components basic render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "rgba(0,0,0,0.08)",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": 718,
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          caretHidden={true}
          keyboardType="numeric"
          maxLength={6}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 60,
              "opacity": 0,
              "width": 230,
              "zIndex": 0,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "height": 60,
                "justifyContent": "center",
                "marginLeft": 20,
                "position": "absolute",
                "width": 668,
                "zIndex": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgba(51,51,51,0.13)",
                  "fontSize": 20,
                }
              }
            >
              请输入
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "height": 50,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#0076FF",
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#0076FF",
                },
              ]
            }
          >
            随机生成
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`getPassword components passwordColor render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "rgba(0,0,0,0.08)",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": 718,
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          caretHidden={true}
          keyboardType="numeric"
          maxLength={6}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 60,
              "opacity": 0,
              "width": 230,
              "zIndex": 0,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "height": 60,
                "justifyContent": "center",
                "marginLeft": 20,
                "position": "absolute",
                "width": 668,
                "zIndex": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgba(51,51,51,0.13)",
                  "fontSize": 20,
                }
              }
            >
              请输入
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "height": 50,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#0076FF",
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#0076FF",
                },
              ]
            }
          >
            随机生成
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`getPassword components passwordNumber render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "rgba(0,0,0,0.08)",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": 718,
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          caretHidden={true}
          keyboardType="numeric"
          maxLength={6}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 60,
              "opacity": 0,
              "width": 230,
              "zIndex": 0,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "height": 60,
                "justifyContent": "center",
                "marginLeft": 20,
                "position": "absolute",
                "width": 668,
                "zIndex": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgba(51,51,51,0.13)",
                  "fontSize": 20,
                }
              }
            >
              请输入
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "height": 50,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#0076FF",
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#0076FF",
                },
              ]
            }
          >
            随机生成
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`getPassword components should TouchableOpacity event 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "rgba(0,0,0,0.08)",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": 718,
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          caretHidden={true}
          keyboardType="numeric"
          maxLength={6}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 60,
              "opacity": 0,
              "width": 230,
              "zIndex": 0,
            }
          }
          underlineColorAndroid="transparent"
          value="276800"
        />
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "height": 60,
                "justifyContent": "center",
                "marginLeft": 20,
                "position": "absolute",
                "width": 668,
                "zIndex": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#0076FF",
                    "fontSize": 24,
                    "fontWeight": "bold",
                    "letterSpacing": 8,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }
            >
              276800
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "height": 50,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#0076FF",
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#0076FF",
                },
              ]
            }
          >
            随机生成
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
